.github/workflows/build-envoy-images-release.yaml

```diff
@@ -38,6 +38,31 @@ jobs:
         run: |
           echo "${{ github.event.pull_request.head.sha }}" >SOURCE_VERSION
           echo "ENVOY_VERSION=$(cat ENVOY_VERSION)" >> $GITHUB_ENV
+          echo "BAZEL_VERSION=$(cat .bazelversion)" >> $GITHUB_ENV
+          echo "BUILDER_DOCKER_HASH=$(git ls-tree --full-tree HEAD -- ./Dockerfile.builder.tests | awk '{ print $3 }')" >> $GITHUB_ENV
+
+      - name: Checking if cilium-envoy-builder image exists
+        id: cilium-builder-tests-tag-in-repositories
+        shell: bash
+        run: |
+          if docker buildx imagetools inspect quay.io/${{ github.repository_owner }}/cilium-envoy-builder:tests-${{ env.BAZEL_VERSION }}-${{ env.BUILDER_DOCKER_HASH }} &>/dev/null; then
+            echo exists="true" >> $GITHUB_OUTPUT
+          else
+            echo exists="false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: PR Multi-arch build & push of Builder image (dev)
+        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671 # v4.0.0
+        if: steps.cilium-builder-tests-tag-in-repositories.outputs.exists == 'false'
+        id: docker_build_builder_tests_ci
+        with:
+          provenance: false
+          context: .
+          file: ./Dockerfile.builder.tests
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: quay.io/${{ github.repository_owner }}/cilium-envoy-builder:tests-${{ env.BAZEL_VERSION }}-${{ env.BUILDER_DOCKER_HASH }}
```

Comment:
Now that the builder tag is different, it would be nice if `Makefile.docker` would compute the `BUILDER_BASE` same way as in here and pass that in `DOCKER_BUILD_OPTS`. That way we would not need to manually update the Dockerfiles for local builds.